From 53b2b725226e64707acf9481dd2a40e1adaf5103 Mon Sep 17 00:00:00 2001
From: pdeshpa3 <pdeshpa3@binghamton.edu>
Date: Wed, 28 Apr 2021 23:40:40 -0400
Subject: [PATCH] Create Patch File

---
 client3.py     | 11 ++++++++---
 client_test.py |  6 ++++++
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/client3.py b/client3.py
index f1771c3..15dc591 100644
--- a/client3.py
+++ b/client3.py
@@ -35,14 +35,17 @@ def getDataPoint(quote):
 	stock = quote['stock']
 	bid_price = float(quote['top_bid']['price'])
 	ask_price = float(quote['top_ask']['price'])
-	price = bid_price
+	price = (bid_price + ask_price)/2
 	return stock, bid_price, ask_price, price
 
 def getRatio(price_a, price_b):
 	""" Get ratio of price_a and price_b """
 	""" ------------- Update this function ------------- """
 	""" Also create some unit tests for this function in client_test.py """
-	return 1
+	if (price_b == 0):
+		return -1
+	
+	return price_a/price_b
 
 # Main
 if __name__ == "__main__":
@@ -52,8 +55,10 @@ if __name__ == "__main__":
 		quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
 		""" ----------- Update to get the ratio --------------- """
+		prices = {}
 		for quote in quotes:
 			stock, bid_price, ask_price, price = getDataPoint(quote)
+			prices[stock] = price
 			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-		print ("Ratio %s" % getRatio(price, price))
+		print ("Ratio %s" % getRatio(prices['ABC'], prices['DEF']))
diff --git a/client_test.py b/client_test.py
index af2bf26..24171c5 100644
--- a/client_test.py
+++ b/client_test.py
@@ -16,8 +16,14 @@ class ClientTest(unittest.TestCase):
     ]
     """ ------------ Add the assertion below ------------ """
 
+    for quote in quotes:
+      self.assertEqual(getDataPoint(quote), (quote['stock'], quote['top_bid']['price'],quote['top_ask']['price'],(quote['top_bid']['price']+quote['top_ask']['price'])/2))
+
 
   """ ------------ Add more unit tests ------------ """
+  def test_getRatio_calculateRatioPriceZero(self):
+    self.assertEqual(getRatio(121.2,0),-1)
+    self.assertEqual(getRatio(0,121.2),0)
 
 
 
-- 
2.27.0.windows.1

